proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,155 @@
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+        .library(
+            name: "SwiftToolsSupport-auto",
+            targets: ["TSCBasic", "TSCUtility"]),
+
+        .library(
+            name: "TSCTestSupport",
+            targets: ["TSCTestSupport"]),
+        .executable(
+            name: "TSCTestSupportExecutable",
+            targets: ["TSCTestSupportExecutable"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above Package descriptions, the `SwiftToolsSupport` product does not match a target within the package description it must be specified explicitly.
+
+```swift
+@package(url: "https://github.com/apple/swift-tools-support-core.git", products: ["SwiftToolsSupport"]) 
+import SwiftToolsSupport 
+```
+
+#### Dependency version requirements
+
+The `@package(..)` annotation would also allow a developer to specify requirements to ensure your scripts are reproducible. The syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+*Revision*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .revision( "065675b3d1364a6f63b94a9c89be2e9ed0a4c3a1")) 
+import Yams 
+```
+
+*Branch*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .branch("releases/1.0")) 
+import Yams 
+```
+
+*Exact Version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .exact("1.0")) 
+import Yams 
+```
+
+*Range of version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .from("1.0")) 
+import Yams 
+```
+
+*Local dependency*
+Absolute path
+```swift
+@package(path: "/Users/username/Yams") 
+import Yams 
+```
+
+Relative path
+```swift
+@package(path: "dependencies/Yams") 
+import Yams 
+```
+
+### Integration with Swift Compiler 
+
+The `@package()` syntax will need to be added to the AST definition and compiler front-end will require changes to parse this syntax.
+
+### Integration with SwiftPM
+
+#### Execution
+Scripts that utilize the `@package()` attribute can be executed by `swift run` and managed by Swift package manager.
+
+Swift package manager will utilize the changes to the compiler front-end to access packages referenced using the `@package` attribute.
```

Comment:
What changes? Is this referencing something mentioned elsewhere?

Reply:
This was very briefly mentioned above this section:
> The `@package()` syntax will need to be added to the AST definition and the compiler front-end will require changes to parse this syntax.

I didn't call too much attention to it in the initial proposal besides stating it needs to be done and the details would largely be debated in the implementation PRs. The idea was largely that we'll need changes to the front-end for parsing the `@package` attribute and also likely new flags for the Swift driver to return dependencies to SwiftPM.